test/distributed/test_c10d_nccl.py

```diff
@@ -500,6 +500,25 @@ def reduce_scatter(outputs, input_lists, op):
             # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
             self.assertEqualIgnoreType(expected, output[i])
 
+    @requires_nccl()
+    def test_reduce_scatter_base_ops(self):
+        store = c10d.FileStore(self.file.name, self.world_size)
+        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+
+        def reduce_scatter_base(output_t, input_t):
+            work = pg._allgather_base(output_t, input_t)
```

Comment:
it is testing _allgather_base, not _reduce_scatter_base?

also, would you please add a  test for dist._reduce_scatter_base() API as well? and the error cases

Reply:
sg!

Reply:
why is '_allgather_base' called here?